### DISCUSSION TITLE
Help me with Vitepress

### DISCUSSION THREAD

**Discussion by MikeCZ23** [element-plus/element-plus#16763] [Q&A]
ERROR: No loader is configured for ".vue" files
or help me with installation on Vitepress.

EDIT: okay, i think solved, but
---

How i can switch theme? dark/light (css); import all css lib

how can I incorporate Component?
e.g.: Lorem impum ``<button />`` lorem impum
or
Lorem impum ``<tag />`` lorem impum
etc.

```ts
//.vitepress/config.mts
import { defineConfig } from 'vitepress'
...
import AutoImport from 'unplugin-auto-import/vite'
import Components from 'unplugin-vue-components/vite'
import { ElementPlusResolver } from 'unplugin-vue-components/resolvers'

export default defineConfig({ 
vite: {
    plugins: [
      AutoImport({
        include: [/\.vue$/, /\.vue\?vue/, /\.md$/],
        resolvers: [ElementPlusResolver()],
      }),    
      Components({
        dirs: ['components'],
        include: [/\.vue$/, /\.vue\?vue/, /\.md$/],
        resolvers: [ElementPlusResolver()]
      }),
    ],
    ssr: { noExternal: ['element-plus'] },
    css: {
      postcss: {
        plugins: [
          {
            postcssPlugin: 'internal:charset-removal',
            AtRule: {
              charset: (atRule) => {
                if (atRule.name === 'charset') {
                  atRule.remove();
                }
              }
            }
          }
        ]
      }
    },
...
```
```ts
//.vitepress/theme/index.ts
import type { Theme } from "vitepress";
import DefaultTheme from "vitepress/theme"
import './custom.css';
...
import PlButton from '../../docs/readme/test/theme/components/PlButton.vue';
import 'element-plus/dist/index.css'

export default {
...
extends: DefaultTheme,
enhanceApp({ app }) {
...
app.component("PlButton", PlButton);
}
} satisfies Theme;
```
icons can't be imported at all. I don't know how

**Comment by warmthsea** [comment]
reference:
https://github.com/element-plus/element-plus/tree/dev/docs

**Reply by MikeCZ23** [reply]
I'm not wise to that.
vite.config.ts no work for me

**Comment by MikeCZ23** [comment]
I have a different problem now. somehow I managed to import it, but the .css component libraries don't work much and the components only work somewhere - /news/ yes, index.md yes, /readme/ no

but, on localhost all works fine (except css libraries). on github no way

**Reply by warmthsea** [reply]
If you have access to this address, there should be examples of vitepress + elementplus above.
https://www.vuetelescope.com/explore?vueVersion_gte=3&framework.slug=vitepress&_sort=lastDetectedAt:desc

But I can't seem to open it recently.
https://github.com/nuxtlabs/vue-telescope-analyzer/issues/109

**Reply by MikeCZ23** [reply]
hmm page no open, but browser extension say everythink ok

**Comment by xing403** [comment]
我这里在`vitepress`中使用了组件库 [看这里](https://github.com/xing403/blog/blob/master/.vitepress/theme/index.ts) ，你可以使用这种方式进行引入

**Reply by MikeCZ23** [reply]
ok, I think I solved the problem with /readme/ not working. but the problem with css - eg.: the calendars have a strange css different than the one on this project page.
I don't know how to insert components

**Comment by MikeCZ23** [comment]
you can look [web](https://github.com/MikeCZ23/mikecz23.github.io/tree/main/.vitepress)
